runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -1017,21 +1294,229 @@ MM_SchedulingDelegate::calculateEdenSize(MM_EnvironmentVLHGC *env)
+	 */
+	intptr_t edenChange = (intptr_t)recommendedEdenSizeBytes - currentIdealEdenBytes;
+	intptr_t targetEdenChange = (intptr_t)(edenChange * edenChangeSpeed);
+	uintptr_t targetEdenBytes = currentIdealEdenBytes + targetEdenChange;
+	uintptr_t targetEdenRegions = targetEdenBytes / _regionManager->getRegionSize();
+
+	_edenSizeFactor = (intptr_t)targetEdenRegions - currentIdealEdenRegions;
+}
+
+void
+MM_SchedulingDelegate::checkEdenSizeAfterPgc(MM_EnvironmentVLHGC *env, bool globalSweepHappened)
+{
+	if (!_extensions->statupPhaseFinished) {
+		/* Don't change eden size during startup phase - keep it at default */
+		return;
+	}
+
+	if (_currentlyPerformingGMP && !globalSweepHappened) {
+		/* 
+		 * Don't change eden size while GMP cycle is running - 
+		 * Unless a global sweep just happened, in which case we allow eden to change on first PGC after GMP 
+		 */
+		return;
+	}
+
+	if (heapIsFullyExpanded(env)) {
+		/* 
+		 * When heap is full, try to get eden to such a size to minimize overhead, while keeping into account the limits imposed by
+		 * the heap size, and how much free space is available, since "blindly" increasing eden is no longer an option
+		 */
+		if (globalSweepHappened) {
+			/* 
+			 * Take a more aggreessive step towards ideal eden. 
+			 * At this point we have the most accuate information about liveness in the heap, so we make the most informed decision
+			 */
+			moveTowardRecommendedEden(env, 0.5);
+			resetPgcTimeStatistics(env);
+		} else if ((_pgcCountSinceGMPEnd & (consecutivePGCToChangeEden - 1)) == 0) {
+			/* Every consecutivePGCToChangeEden number of pgc's, re-evaluate eden size, and move towards it */
+			moveTowardRecommendedEden(env, 0.25);
+		}
+	} else if (0 == _pgcCountSinceGMPEnd % 3) {
+		/*
+		 * On every third pgc, make an adjustment to eden size based on observed pgc values. 
+		 * Waiting for every third PGC, allows some statistics (which are being averaged) to settle to their new true averages
+ 		 */
+
+		intptr_t edenRegionChange = 0;
+		intptr_t edenChangeMagnitude = (intptr_t)ceil((0.1 * getIdealEdenSizeInBytes(env)) / _regionManager->getRegionSize());
+
+		double hybridEdenOverhead = calculateHybridEdenOverhead(env, _historicalPartialGCTime, _partialGcOverhead);
+		
+		/* 
+		 * Aim to get hybrid PGC overhead between extensions->dnssExpectedTimeRatioMinimum and extensions->dnssExpectedTimeRatioMaximum 
+		 * by increasing or decreasing eden by 10% 
+		 */
+		if (_extensions->dnssExpectedTimeRatioMinimum._valueSpecified > hybridEdenOverhead ) {
+			/* Shrink eden a bit */
+			edenRegionChange = edenChangeMagnitude * -1;
+		} else if (_extensions->dnssExpectedTimeRatioMaximum._valueSpecified < hybridEdenOverhead) {
+			/* Expand eden a bit */
+			edenRegionChange = edenChangeMagnitude;
+		}
+
+		_edenSizeFactor += edenRegionChange;	
+	}
+}
+
+double 
+MM_SchedulingDelegate::mapPgcTimeToPgcOverhead(MM_EnvironmentVLHGC *env, uintptr_t partialGcTimeMs) {
```

Comment:
I'd rename to mapPgcPauseOverheadToPgcCPUOverhead. The key change was to use 'pause' instead of 'time' what slightly better represents what we are trying to optimize (that is meet pause threshold) - 'time' somewhat implies we are trying to optimize CPU overhead.

Reply:
consequnetly, I'd adjust parameter name and comments below to use 'pause'